portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -107,25 +110,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the converted input from form.
-    Comment newComment = getCommentFromForm(request);
-    // Add it to the comment history.
+    // Add new comment to the comment history.
     Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("username", newComment.getUsername());
-    commentEntity.setProperty("text", newComment.getText());
-    commentEntity.setProperty("date", newComment.getDate());
-    commentEntity.setProperty("timestamp", newComment.getTimestamp());
+    UserService userService = UserServiceFactory.getUserService();
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    /* The entity has a field called username. This field will keep the email
+    or the nickname. If the user doesn't have a problem with his email be
+    displayed near his comments and doesn't set any nickname, then username
+    = email. Otherwise, if he has a nickname, then username = nickname. */ 
```

Comment:
Optional: It might be a good idea to store the user id and look up the nickname (if one is set) on demand whenever you are returning the comments.

This allows the user to actually change their nickname, and have the change applied retroactive across all comments (I think that's what the majority of the webapps I've seen do). This also allows the user to change their minds (oh damn, I didn't want my email shown, let me set a nickname now), which is a good privacy tenant in general.

Not sure if the exercise suggests otherwise though.

Reply:
Yes that is a good idea, I will add it soon.

Reply:
Done